packages/shared/src/utils/proxy.ts

```diff
@@ -20,3 +20,13 @@ export function proxyUrlToAbsoluteURL(url: string | undefined): string {
   }
   return isProxyUrlRelative(url) ? new URL(url, window.location.origin).toString() : url;
 }
+
+export function createProxyUrl({ request, relativePath }: { request: Request; relativePath: string }): string {
+  const { headers, url: initialUrl } = request;
+  const url = new URL(initialUrl);
+  const host = headers.get('X-Forwarded-Host') ?? headers.get('host') ?? url.host;
+
+  // X-Forwarded-Proto could be 'https, http'
+  const protocol = headers.get('X-Forwarded-Proto')?.split(',')[0] ?? url.protocol;
+  return new URL(relativePath, `${protocol}://${host}`).toString();
+}
```

Comment:
:upside_down_face: Wonder if we should keep this out of shared until we have an indication that it can be re-usable.

Unless we simply remove the duplication from nextjs once we can use the shared module.

Reply:
My plan is to simply import the utility from `@clerk/shared` and delete the duplicated part once the bundling issue is resolved.